src/sentry/static/sentry/app/views/settings/organizationTeams/teamDetails.jsx

```diff
@@ -62,15 +61,62 @@ const TeamDetails = createReactClass({
     });
   },
 
+  handleRequestAccess() {
+    const {api, params} = this.props;
+    const {team} = this.state;
+
+    if (!team) {
+      return;
+    }
+
+    this.setState({
+      requesting: true,
+    });
+
+    joinTeam(
+      api,
+      {
+        orgId: params.orgId,
+        teamId: team.slug,
+      },
+      {
+        success: () => {
+          addSuccessMessage(
+            tct('You have requested access to [team]', {
+              team: `#${team.slug}`,
+            })
+          );
+          this.setState({
+            requesting: false,
+          });
+        },
+        error: () => {
+          addErrorMessage(
+            tct('Unable to request access to [team]', {
+              team: `#${team.slug}`,
+            })
+          );
+          this.setState({
+            requesting: false,
+          });
+        },
+      }
+    );
+  },
+
   fetchData() {
+    this.setState({
+      loading: true,
+      error: false,
+    });
     fetchTeamDetails(this.props.api, this.props.params);
   },
 
   onTeamChange(data) {
     const team = this.state.team;
     if (data.slug !== team.slug) {
       const orgId = this.props.params.orgId;
-      browserHistory.push(`/organizations/${orgId}/teams/${data.slug}/settings/`);
+      browserHistory.replace(`/organizations/${orgId}/teams/${data.slug}/settings/`);
```

Comment:
If slug gets updated, we want to navigate to new slug w/o a new history entry so if user presses back it won't take them to invalid team details.